### PR TITLE
chore: update rockcraft project, add sanity test, and tox.ini file

### LINKED ISSUE #71: Update the `centraldashboard` rockcraft project
### Context

 Update the corresponding rockcraft.yaml file to make sure the version of CKF 1.8/1.9 is built.

### What needs to get done

The rockcraft.yaml is updated
There is automation that builds the image in each PR
There is automation that pushes the image, once a PR is merged

### Definition of Done

To make sure we have less CVEs in 1.8/1.9 than upstream, and enable us to be able to potentially create patches for package versions in the future

### PR DESCRIPTION
**Pull request by DnPlas**
This commit updates the rockcraft project to match the new requirements from the latest rockcraft version. It also refactors the security requirement to match the latest format in other rock repositories.
Bash is now removed from the image as it is not required and not recommended in production environments, and the build dependency on apt is also removed as it was not used at all. Adding sanity tests in this rock's repository and its corresponding tox file.

Fixes #71

#### Testing instructions

Prerequisites:
* skopeo
* docker

1. Build the rock `rockcraft pack -v`
2. Save the rock as an OCI image in docker's local registry `skopeo --insecure-policy copy oci-archive:kubeflow-central-dashboard_1.7_amd64.rock docker-daemon:cdash:0.1`
3. Run a container with the image and replace the entrypoint with a shell `docker run --rm -ti --entrypoint=/bin/sh cdash:0.1`
4. Ensure that:

* The `/app` directory exists
* The permissions of the `/app` directory are `drwxr-xr-x  11 _daemon_ _daemon_ 4.0K Feb  4 00:08 app`
* You can run the command `/usr/bin/npm start --prefix /app` and it outputs the following message:
```
Using Profiles service at http://localhost:8081/kfam
Server listening on port http://localhost:8082 (in development mode)
```

### CODE REVIEW

**Comment by DnPlas** [comment]
The CI of this PR is currently failing because of https://github.com/canonical/charmed-kubeflow-workflows/pull/45. We must re-run everything once that gets merged.

File: centraldashboard/rockcraft.yaml
Comment (kimwnasptd) [inline centraldashboard/rockcraft.yaml:7]: Why is the version 1.7, and getting merged to main, if the goal of the task is to have this in 1.8?

File: centraldashboard/rockcraft.yaml
Comment (orfeas-k) [inline centraldashboard/rockcraft.yaml:16]: Is there a reason for changing this?

File: centraldashboard/rockcraft.yaml
Comment (DnPlas) [reply centraldashboard/rockcraft.yaml:7]: @kimwnasptd you are right in that we want to upgrade for 1.8, I must've been focused on updating the rock and making it work that I forgot the version. Will update. 

@misohu I will have your suggestion in mind, thanks!

File: centraldashboard/rockcraft.yaml
Comment (DnPlas) [reply centraldashboard/rockcraft.yaml:55]: Oh, nice suggestion. We don't really follow a standard, but I'd do it this way. Will update.

File: centraldashboard/rockcraft.yaml
Comment (DnPlas) [reply centraldashboard/rockcraft.yaml:16]: I don't think we follow a standard, and since I was following another example I had, I left it like that. We should probably discuss how we want to approach the `override` strategy, until then, should I keep `replace`? Wdyt?

File: centraldashboard/rockcraft.yaml
Comment (DnPlas) [reply centraldashboard/rockcraft.yaml:7]: @misohu about the node version, [this commit](https://github.com/canonical/kubeflow-rocks/commit/82e3850f8d6baffc8d96d4bf7fa42905bf4d30f7) introduced the change, but there is no apparent justification. I am setting it to node 14 to keep consistency with upstream.
About the user, unfortunately it is a requirement by the security team and I don't think we can change it from __daemon__ to node, and tbf I don't think there will be an issue.

File: centraldashboard/rockcraft.yaml
Comment (DnPlas) [reply centraldashboard/rockcraft.yaml:55]: After trying to use this, it seems like canonical/craft-parts#427 is still valid, causing the working dir to not be set correctly. I will leave the `cd` for now, but the suggestion was good.

File: centraldashboard/rockcraft.yaml
Comment (misohu) [reply centraldashboard/rockcraft.yaml]: yeah we should always update the node to latest possible for critical vulnerabilities ... yeah the user makes sense if its not doing problems looking good

File: centraldashboard/rockcraft.yaml
Comment (orfeas-k) [reply centraldashboard/rockcraft.yaml]: Makes sense. I added a [note](https://docs.google.com/document/d/1Zm11x2sRxue3sUEWe776-6ZHPwyDl6hVXP41z2zhCno/edit#heading=h.6nwxcci67ohi) for the next how do you rock meeting, so we can discuss it as a team.

**Review by misohu — APPROVED** [review]

**Pull request merged**
